From: Florian Fainelli <f.fainelli@gmail.com>
Subject: Add myloader partition table parser

[john@phozen.org: shoud be upstreamable]

lede-commit: d8bf22859b51faa09d22c056fe221a45d2f7a3b8
Signed-off-by: Florian Fainelli <f.fainelli@gmail.com>
---
 drivers/mtd/Kconfig  | 16 ++++++++++++++++
 drivers/mtd/Makefile |  1 +
 2 files changed, 17 insertions(+)

--- a/drivers/mtd/Kconfig
+++ b/drivers/mtd/Kconfig
@@ -177,6 +177,22 @@ menu "Partition parsers"
 source "drivers/mtd/parsers/Kconfig"
 endmenu
 
+config MTD_MYLOADER_PARTS
+	tristate "MyLoader partition parsing"
+	depends on ADM5120 || ATH25 || ATH79
+	---help---
+	  MyLoader is a bootloader which allows the user to define partitions
+	  in flash devices, by putting a table in the second erase block
+	  on the device, similar to a partition table. This table gives the 
+	  offsets and lengths of the user defined partitions.
+
+	  If you need code which can detect and parse these tables, and
+	  register MTD 'partitions' corresponding to each image detected,
+	  enable this option.
+
+	  You will still need the parsing functions to be called by the driver
+	  for your particular device. It won't happen automatically.
+
 comment "User Modules And Translation Layers"
 
 #
--- a/drivers/mtd/Makefile
+++ b/drivers/mtd/Makefile
@@ -16,6 +16,7 @@ obj-$(CONFIG_MTD_AFS_PARTS)	+= afs.o
 obj-$(CONFIG_MTD_AR7_PARTS)	+= ar7part.o
 obj-$(CONFIG_MTD_BCM63XX_PARTS)	+= bcm63xxpart.o
 obj-$(CONFIG_MTD_BCM47XX_PARTS)	+= bcm47xxpart.o
+obj-$(CONFIG_MTD_MYLOADER_PARTS) += myloader.o
 obj-y				+= parsers/
 
 # 'Users' - code which presents functionality to userspace.
